elastic/kibana
# [Infrastructure UI] Lens Embeddable attributes builder refactor

relates to [#160381](https://github.com/elastic/kibana/issues/160381)
closes [#161432](https://github.com/elastic/kibana/issues/161432)

## Summary

This pull request aims to enhance the usage of Lens Embeddable by introducing abstractions that simplify the creation of the necessary data structures for rendering charts. The goal is to improve the DX by providing clearer and more intuitive interfaces.

More details in the [README.md](https://github.com/elastic/kibana/blob/62a9ef70e6bbbb6798bd78083dac097d84b85ec5/x-pack/plugins/infra/public/common/visualizations/lens/README.md)

### For reviewers

- The majority of the changes are concentrated in the `common/visualizations` directory. The `formulas/host` files have been modified to only contain the formula itself. The formulas are now designed to be independent of specific chart styles.

- The `use_lens_attributes` hook has been modified to receive the essential information. Previously, it had knowledge of host formulas, but now it accepts an abstracted chart representation and returns the JSON object accordingly.

- Chart-specific style configurations have been moved to the chart usage component in the `tile` and `metric_chart` components.


### How to test 

- Start a local Kibana instance
- Navigate to `Infrastructure` > `Hosts`
- Check all charts and navigate to Lens.
